core/src/main/kotlin/configuration.kt

```diff
@@ -65,55 +74,50 @@ fun DokkaConfigurationImpl(json: String): DokkaConfigurationImpl = parseJson(jso
 fun DokkaConfiguration.toJsonString(): String = toJsonString(this)
 
 interface DokkaConfiguration : Serializable {
-    val outputDir: String
-    val cacheRoot: String?
+    val outputDir: File
+    val cacheRoot: File?
     val offlineMode: Boolean
     val failOnWarning: Boolean
     val sourceSets: List<DokkaSourceSet>
     val modules: List<DokkaModuleDescription>
-    val pluginsClasspath: List<File>
+    val pluginsClasspath: Set<File>
     val pluginsConfiguration: Map<String, String>
 
     interface DokkaSourceSet : Serializable {
         val sourceSetID: DokkaSourceSetID
         val displayName: String
         val moduleDisplayName: String
-        val classpath: List<String>
-        val sourceRoots: List<SourceRoot>
+        val classpath: Set<File>
```

Comment:
I know that it shouldn't matter in the perfect world, but sadly most developers at least once encountered a project that depended on the order of compiler classpath. If we treat classpath as a set we can introduce unresolved types in dokka, for projects that are compiling and working.
I think for this one field we should return to the interface that requires ordering.

Reply:
Pushed a commit to change from `Set` to `List`.